File: app/src/main/java/ru/vpiska/helper/maps/DatePicker.java
package ru.vpiska.helper.maps;
import android.annotation.SuppressLint;
import android.app.AlertDialog;
import android.app.DatePickerDialog;
import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.v4.app.DialogFragment;

import android.widget.Button;
import android.widget.TextView;

import java.text.SimpleDateFormat;
import java.util.Calendar;

import ru.vpiska.R;

public class DatePicker extends DialogFragment
        implements DatePickerDialog.OnDateSetListener {


    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        // определяем текущую дату


        final Calendar c = Calendar.getInstance();
        int year = c.get(Calendar.YEAR);
        int month = c.get(Calendar.MONTH);
        int day = c.get(Calendar.DAY_OF_MONTH);


        // создаем DatePickerDialog и возвращаем его
        DatePickerDialog picker = new DatePickerDialog(getActivity(), AlertDialog.THEME_HOLO_LIGHT, this,
                year, month, day);

        picker.setTitle(getResources().getString(R.string.choose_date));
        c.set(year+1, month, day);//Year,Mounth -1,Day
        long now = System.currentTimeMillis() - 1000;

        picker.getDatePicker().setMinDate(now);

        picker.getDatePicker().setMaxDate(c.getTimeInMillis());


        return picker;
    }

    @Override
    public void onStart() {
        super.onStart();
        // добавляем кастомный текст для кнопки
        Button nButton = ((AlertDialog) getDialog())
                .getButton(DialogInterface.BUTTON_POSITIVE);
        nButton.setText(getResources().getString(R.string.ready));
        Button ngButton = ((AlertDialog) getDialog())
                .getButton(DialogInterface.BUTTON_NEGATIVE);
        ngButton.setText(getResources().getString(R.string.cancel));

    }

    @Override
    public void onDateSet(android.widget.DatePicker datePicker, int year,
                          int month, int day) {

        final Calendar c = Calendar.getInstance();
        @SuppressLint("SimpleDateFormat") SimpleDateFormat f = new SimpleDateFormat("yyyy-MM-dd");
        c.set(year, month+1, day);
        String timestamp = f.format(c.getTime());


        TextView txtDate = (TextView) getActivity().findViewById(R.id.txtDate);
        txtDate.setText(timestamp);
    }
}

File: app/src/main/java/ru/vpiska/helper/maps/DatePicker.kt
package ru.vpiska.helper.maps

import android.app.AlertDialog
import android.app.DatePickerDialog
import android.app.Dialog
import android.content.DialogInterface
import android.os.Bundle
import android.support.v4.app.DialogFragment
import android.widget.TextView
import ru.vpiska.R
import java.text.SimpleDateFormat
import java.util.*

class DatePicker : DialogFragment(), DatePickerDialog.OnDateSetListener {


    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        // определяем текущую дату


        val c = Calendar.getInstance()
        val year = c.get(Calendar.YEAR)
        val month = c.get(Calendar.MONTH)
        val day = c.get(Calendar.DAY_OF_MONTH)


        // создаем DatePickerDialog и возвращаем его
        val picker = DatePickerDialog(activity!!, AlertDialog.THEME_HOLO_LIGHT, this,
                year, month, day)

        picker.setTitle(resources.getString(R.string.choose_date))
        c.set(year + 1, month, day)
        val now = System.currentTimeMillis() - 1000

        picker.datePicker.minDate = now

        picker.datePicker.maxDate = c.timeInMillis


        return picker
    }

    override fun onStart() {
        super.onStart()
        // добавляем кастомный текст для кнопки
        val nButton = (dialog as AlertDialog)
                .getButton(DialogInterface.BUTTON_POSITIVE)
        nButton.text = resources.getString(R.string.ready)
        val ngButton = (dialog as AlertDialog)
                .getButton(DialogInterface.BUTTON_NEGATIVE)
        ngButton.text = resources.getString(R.string.cancel)

    }

    override fun onDateSet(datePicker: android.widget.DatePicker, year: Int,
                           month: Int, day: Int) {

        val c = Calendar.getInstance()
        val f = SimpleDateFormat("yyyy-MM-dd")
        c.set(year, month + 1, day)
        val timestamp = f.format(c.time)


        val txtDate = activity!!.findViewById<TextView>(R.id.txtDate)
        txtDate.text = timestamp
    }
}